### ISSUE TITLE
Incorrect validation when calling super in ParameterizedFunctions

### ISSUE THREAD

**Issue by jlstevens** [holoviz/param#455]
I would expect the following pattern to work, where the baseclass generalizes the set of allowable parameter values while a subclass has a stricter definition:

```python
import param

class Foo(param.ParameterizedFunction):
    a = param.ClassSelector(default=['foo'], class_=(dict, list))
    def __call__(self, **params):
        self.p = param.ParamOverrides(self, params)
        return 1
    
class Bar(Foo):
    a = param.List(default=['bar'])
    def __call__(self, **params):
        return super().__call__(**params) + 1
    
Bar.instance(a=[4])
```

What seems to be happening is that `class_=(dict, list)` is being used by `List` to validate that the elements of the list are lists or dictionaries (which is obviously wrong).

**Comment by jbednar** [comment]
That does seem like. a bug, but it's also illustrating bad practice to the extent that parameters are considered part of an object's declaration, since Bar does not satisfy "is a" Foo (since it doesn't accept all values of 'a' that Foo does).

**Comment by jlstevens** [comment]
> it's also illustrating bad practice ... since Bar does not satisfy "is a" Foo (since it doesn't accept all values of 'a' that Foo does).

I agree that if you were to write `Bar` given the above definition of `Foo` you would match the declarations. 

The use case described in the PR above is when you generalize a base class and you don't know about existing users who may have written subclasses before that generalization was made. Those people should not have their subclasses broken as long as they specify values that are an allowed subset of the base class (but a warning would be appropriate, suggesting that they should update to the generalized declaration).

Even if we were to be dogmatic about not allowing subclasses to be more specific, the current behavior is just confusing.



In the HoloViews use case,
